db/migrations/48_remove_groups_items_propagate_access.sql

```diff
@@ -0,0 +1,278 @@
+    IF NEW.version <> OLD.version THEN
+        SET @curVersion = NEW.version;
+    ELSE
+        SELECT (UNIX_TIMESTAMP() * 10) INTO @curVersion;
+    END IF;
+    IF NOT (OLD.`id` = NEW.`id` AND OLD.`group_id` <=> NEW.`group_id` AND OLD.`item_id` <=> NEW.`item_id` AND
+            OLD.`creator_user_id` <=> NEW.`creator_user_id` AND OLD.`partial_access_date` <=> NEW.`partial_access_date` AND
+            OLD.`full_access_date` <=> NEW.`full_access_date` AND OLD.`access_reason` <=> NEW.`access_reason` AND
+            OLD.`access_solutions_date` <=> NEW.`access_solutions_date` AND OLD.`owner_access` <=> NEW.`owner_access` AND
+            OLD.`manager_access` <=> NEW.`manager_access` AND OLD.`cached_partial_access_date` <=> NEW.`cached_partial_access_date` AND
+            OLD.`cached_full_access_date` <=> NEW.`cached_full_access_date` AND OLD.`cached_access_solutions_date` <=> NEW.`cached_access_solutions_date` AND
+            OLD.`cached_grayed_access_date` <=> NEW.`cached_grayed_access_date` AND OLD.`cached_full_access` <=> NEW.`cached_full_access` AND
+            OLD.`cached_partial_access` <=> NEW.`cached_partial_access` AND OLD.`cached_access_solutions` <=> NEW.`cached_access_solutions` AND
+            OLD.`cached_grayed_access` <=> NEW.`cached_grayed_access` AND OLD.`cached_manager_access` <=> NEW.`cached_manager_access`) THEN
+        SET NEW.version = @curVersion;
+        UPDATE `history_groups_items` SET `next_version` = @curVersion WHERE `id` = OLD.`id` AND `next_version` IS NULL;
+        INSERT INTO `history_groups_items` (
+                `id`,`version`,`group_id`,`item_id`,`creator_user_id`,`partial_access_date`,`full_access_date`,`access_reason`,
+                `access_solutions_date`,`owner_access`,`manager_access`,`cached_partial_access_date`,`cached_full_access_date`,
+                `cached_access_solutions_date`,`cached_grayed_access_date`,`cached_full_access`,`cached_partial_access`,
+                `cached_access_solutions`,`cached_grayed_access`,`cached_manager_access`
+            ) VALUES (
+                NEW.`id`,@curVersion,NEW.`group_id`,NEW.`item_id`,NEW.`creator_user_id`,NEW.`partial_access_date`,
+                NEW.`full_access_date`,NEW.`access_reason`,NEW.`access_solutions_date`,NEW.`owner_access`,
+                NEW.`manager_access`,NEW.`cached_partial_access_date`,NEW.`cached_full_access_date`,
+                NEW.`cached_access_solutions_date`,NEW.`cached_grayed_access_date`,NEW.`cached_full_access`,
+                NEW.`cached_partial_access`,NEW.`cached_access_solutions`,NEW.`cached_grayed_access`,
+                NEW.`cached_manager_access`
+            );
+    END IF;
+END
+-- +migrate StatementEnd
+DROP TRIGGER IF EXISTS `after_update_groups_items`;
+-- +migrate StatementBegin
+CREATE DEFINER=`algorea`@`%` TRIGGER `after_update_groups_items` AFTER UPDATE ON `groups_items` FOR EACH ROW BEGIN
```

Comment:
These trigger names are quite bad names, they do not define what the function does but when it is run, which already present in the trigger metadata. I don't know if it make sense to do it from now on (for new ones). A bit related with my other comments about comments,  maybe if it has a good name, it does not require comments.

Reply:
For me, `after_update_groups_items` is a perfect name. It really describes the trigger (no matter what it does inside).

Reply:
Also, note that in AlgoreaPlatform these triggers are generated by the PHP code.

Reply:
http://www.dba-oracle.com/t_trigger_naming_standards.htm

Reply:
Actually people use this weird naming for triggers because it was impossible to have more than one trigger for the same table and the same event before MySQL 5.7.2. Although it is possible with 5.7.2+, the tradition persists. I would keep the naming just for the sake of consistency (all our triggers are named in that way).